runtime/cmake/platform/toolcfg/xlc.cmake

```diff
@@ -43,5 +43,11 @@ if(CMAKE_C_COMPILER_IS_XLCLANG)
 	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -fno-rtti)
 else()
 	# xlc/xlc++ options
-	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -g)
+	if(OMR_OS_ZOS)
+		# Specifying -g on z/OS inhibits compiler optimizations.
+		# Instead we use these flags to info required for DDR without hindering the optimizer.
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS -qdebug=nohook -qxplink=noback)
```

Comment:
It appears xplink and noback are the defaults for 64-bit, is specifying it necessary?

I'm not sure `-qdebug=nohook` is desirable. I believe -O3 defaults to `hook=none:profile`

profile - It is the same as specifying call and func. It is the ideal setting for tracing the program with the Performance Analyzer.

@fjeremic @joran any comments?

Reply:
Specifying `noback` does seem to be necessary. Based on experimentation, it appears that enabling debug info un-sets the `noback` option (presumably to aid in debugging inlined functions). I'll look into the hook option

Reply:
Output from `-qlist` 

`-O3`:
```
: *XPLINK(NOBACKCHAIN,NOSTOREARGS,NOCALLBACK,GUARD,OSCALL(NOSTACK))
: *NODEBUG
```

`-O3 -g`:
```
: *XPLINK(BACKCHAIN,STOREARGS,NOCALLBACK,GUARD,OSCALL(NOSTACK))
: *DEBUG(FORMAT(DWARF),LEVEL(0),HOOK(LINE,BLOCK,PATH,CALL,FUNC),SYMBOL,
:        FILE(./test.dbg))
```

`-O3 -qdebug=nohook -qxplink=noback`:
```
: *XPLINK(NOBACKCHAIN,NOSTOREARGS,NOCALLBACK,GUARD,OSCALL(NOSTACK))
: *DEBUG(FORMAT(DWARF),LEVEL(0),NOHOOK,SYMBOL,
:        FILE(./test.dbg))
```

Reply:
Based on that I'm willing to merge, but will wait to see if there are other comments first. We will of course need to perf test the cmake builds to ensure they are equivalent.

Reply:
Sorry no comments for me as I'm not to familiar with this area myself.

Reply:
I'm not familiar with these options either -- but have passed this PR to C/C++ perf SME to review the various nohook/noback options.

Reply:
This looks reasonable to me assuming there's sufficient information captured in the `.dbg` files to satisfy DDR.

Reply:
C/C++ perf SME reviewed, and he "would agree the change seems reasonable to me...I can’t see how this is going to be worse than specifying `-g`, before the change."  I agree we need to perf test to validate this change.